mira/modeling/petri.py

```diff
@@ -96,9 +96,14 @@ def __init__(self, model: Model):
                     'mira_concept': var.concept.json(),
                 }
             }
-            initial = var.data.get('initial_value')
-            if initial is not None:
-                state_data['concentration'] = initial
+            initial_expr = var.data.get('expression')
+            # In this case, initial.expression.args[0] isn't a primitive data type but of types such as
+            # 'One',"Two','Zero'
+            if initial_expr is not None:
+                try:
+                    state_data['concentration'] = float(str(initial_expr))
```

Comment:
Same as above: ideally we'd first try to substitute into the initial_expr to obtain a specific number and then just use that number.

Reply:
Would this line work instead, `float(initial_expr.subs(initial_expr,initial_expr.args[0]))`? This line operates under the assumption that we will only have one free symbol (i.e. the sympy expression that represents the state_data's concentration).